include/irr/asset/ICPUTexture.h

```diff
@@ -187,16 +175,7 @@ class ICPUTexture : public IAsset
     }
     inline void establishFmt()
     {
-        asset::E_FORMAT fmt = m_mipmaps[0]->getColorFormat();
-        for (uint32_t i = 1u; i < m_mipmaps.size(); ++i)
-        {
-            if (fmt != m_mipmaps[i]->getColorFormat())
-            {
-                fmt = EF_UNKNOWN;
-                break;
-            }
-        }
-        m_colorFormat = fmt;
+        m_colorFormat = (*std::find_if(m_mipmaps.begin(), m_mipmaps.end(), [](CImageData* mip) { return mip->getColorFormat() != EF_UNKNOWN; }))->getColorFormat();
```

Comment:
what about the check that all the mips that don't have format `EF_UNKNOWN`, must have the same format?

Reply:
Since constructor (and those init functions as well) is protected, construction code assumes that `create()` already make sure that input `core::vector<asset::CImageData*>` is perfectly valid

Reply:
ok so this sort of validation takes place somewhere else?